vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,265 @@
+                      comment = "#>",
+                      dpi = 300
+                      )
+
+```
+
+## Overview
+
+Branching processes can be used to project infectious disease trends provided 
+we can characterize the distribution of times between 
+successive cases (serial interval), and the distribution of secondary cases 
+produced by a single individual (offspring distribution). Such simulations can be achieved in `bpmodels` with the `chain_sim()` function. @Pearson2020, and 
+@abbott2020 illustrate its application to COVID-19. 
+
+The purpose of this vignette is to use early data on COVID-19 in South Africa [@marivate2020] to illustrate how `bpmodels` can be used to forecast an outbreak. 
+
+
+Let's load the required packages
+
+```{r packages, include=TRUE}
+library("bpmodels")
+library('dplyr')
+library('ggplot2')
+library('lubridate')
+```
+
+### The data
+
+We will get and clean the first $15$ days of the COVID-19 
+outbreak in South Africa to seed the simulation for this example.
+
+```{r data, message=FALSE}
+data_url <- 'https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_timeline_confirmed.csv'
+
+#Read the data in using the url
+covid19_sa <- read.csv(data_url)
+
+# Subset the first 15 days and count the number of cases per date
+covid19_sa <- covid19_sa %>%
+  dplyr::select(date) %>%
+  dplyr::mutate(date = lubridate::dmy(date)) %>%
+  dplyr::filter(date <= lubridate::dmy("20-03-2020")) %>%
+  dplyr::group_by(date) %>%
+  dplyr::summarise(cases = n()) %>%
+  dplyr::ungroup()
+```
+
+### Preparing the inputs  
+
+```{r linelist_gen, message=FALSE}
+days_since_index <- as.integer(covid19_sa$date - min(covid19_sa$date))
+
+start_times <- unlist(mapply(
+  function(x, y) rep(x, times = ifelse(y == 0, 1, y)),
+  days_since_index,
+  covid19_sa$cases
+))
+                       
+```
+
+
+Additionally, `chain_sim()` requires other inputs, which we will specify below: 
+
+```{r input_prep2, message=FALSE}
+#' Date to end simulation (14 day projection in this case)
+projection_window <- 14 # 14 days/ 2-week ahead projection
+
+projection_end_day <- max(days_since_index) + projection_window
+
+#' Number of simulations
+sim_rep <- 100
+
+#' Maximum chain size allowed
+chain_threshold <- 1000
+
+```
+
+#### Serial interval
+
+We also assume based on COVID-19 literature that the 
+serial interval, $si$, is lognormal distributed as follows:
+
+$ E[\text{si}] = \ln \left( \dfrac{\mu^2}{(\sqrt{\mu^2 + \sigma^2}} \right)$
+
+$ SD [\text{si}] = \sqrt {\ln \left(1 + \dfrac{\sigma^2}{\mu^2} \right)}$
+
+with $\mu = 4.7$ and standard deviation $\sigma = 2.9$.
+
+```{r input_prep3, message=FALSE}
+mu <- 4.7
+sigma <- 2.9
+
+si_sd <- sqrt(log(1 + (sigma / mu)^2)) # log standard deviation
+si_mean <- log((mu^2) / (sqrt(sigma^2 + mu^2))) # log mean
+
+#' serial interval function
+serial_interval <- function(sample_size) {
+  si <- rlnorm(sample_size, meanlog = si_mean, sdlog = si_sd)
+  return(si)
+}
```

Comment:
This looks like something that could be added as a small helper function to the package. Even if not, I would suggest explaining this step better with some epi context.

Reply:
I'm more in favour of leaving it as is since it follows naturally from the explanatory text. It's also worth noting that this has been implemented one too many times in other packages ({epiparameter}, {EpiNow2}, etc). I will provide some more explanation.

Reply:
I would suggest using examples from those packages then, it's a good way to show how the packages are connected.

Reply:
You're right but I'm a bit wary of drawing in either package atm.

Reply:
I have now switched to using [epiparameter](https://github.com/epiverse-trace/epiparameter).